File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/configuration/TypeConversionConfiguration.java
package com.jmsoftware.maf.springcloudstarter.configuration;

import cn.hutool.core.date.LocalDateTimeUtil;
import cn.hutool.core.util.StrUtil;
import com.jmsoftware.maf.springcloudstarter.constant.UniversalDateTime;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
import org.springframework.context.annotation.Bean;
import org.springframework.core.convert.converter.Converter;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.time.format.DateTimeFormatter;

/**
 * Description: TypeConvesionConfiguration, change description here.
 *
 * @author Johnny Miller (锺俊), email: johnnysviva@outlook.com, date: 6/27/2021 9:36 PM
 **/
@Slf4j
public class TypeConversionConfiguration {
    @Bean
    public StringToLocalDateConverter localDateConverter() {
        return source -> {
            if (StrUtil.isBlank(source)) {
                return null;
            }
            return LocalDateTimeUtil.parseDate(source, UniversalDateTime.DATE_FORMAT);
        };
    }

    @Bean
    public StringToLocalTimeConverter localTimeConverter() {
        return source -> {
            if (StrUtil.isBlank(source)) {
                return null;
            }
            return LocalTime.parse(source, DateTimeFormatter.ofPattern(UniversalDateTime.TIME_FORMAT));
        };
    }

    @Bean
    @ConditionalOnBean(name = "requestMappingHandlerAdapter")
    public StringToLocalDateTimeConverter localDateTimeConverter() {
        return source -> {
            if (StrUtil.isBlank(source)) {
                return null;
            }
            return LocalDateTimeUtil.parse(source, UniversalDateTime.DATE_TIME_FORMAT);
        };
    }

    interface StringToLocalDateConverter extends Converter<String, LocalDate> {
    }

    interface StringToLocalTimeConverter extends Converter<String, LocalTime> {
    }

    interface StringToLocalDateTimeConverter extends Converter<String, LocalDateTime> {
    }
}

File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/configuration/TypeConversionConfiguration.kt
package com.jmsoftware.maf.springcloudstarter.configuration

import cn.hutool.core.date.LocalDateTimeUtil
import cn.hutool.core.util.StrUtil
import com.jmsoftware.maf.common.constant.UniversalDateTime
import org.springframework.boot.autoconfigure.condition.ConditionalOnBean
import org.springframework.context.annotation.Bean
import org.springframework.core.convert.converter.Converter
import java.time.LocalDate
import java.time.LocalDateTime
import java.time.LocalTime
import java.time.format.DateTimeFormatter

/**
 * # TypeConversionConfiguration
 *
 * Description: TypeConversionConfiguration, change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/14/22 10:58 AM
 */
class TypeConversionConfiguration {
    interface StringToLocalDateConverter : Converter<String?, LocalDate?>
    interface StringToLocalTimeConverter : Converter<String?, LocalTime?>
    interface StringToLocalDateTimeConverter : Converter<String?, LocalDateTime?>

    @Bean
    fun localDateConverter(): StringToLocalDateConverter {
        return object : StringToLocalDateConverter {
            override fun convert(source: String): LocalDate? {
                if (StrUtil.isBlank(source)) {
                    return null
                }
                return LocalDateTimeUtil.parseDate(source, UniversalDateTime.DATE_FORMAT)
            }
        }
    }

    @Bean
    fun localTimeConverter(): StringToLocalTimeConverter {
        return object : StringToLocalTimeConverter {
            override fun convert(source: String): LocalTime? {
                if (StrUtil.isBlank(source)) {
                    return null
                }
                return LocalTime.parse(source, DateTimeFormatter.ofPattern(UniversalDateTime.TIME_FORMAT))
            }
        }
    }

    @Bean
    @ConditionalOnBean(name = ["requestMappingHandlerAdapter"])
    fun localDateTimeConverter(): StringToLocalDateTimeConverter {
        return object : StringToLocalDateTimeConverter {
            override fun convert(source: String): LocalDateTime? {
                if (StrUtil.isBlank(source)) {
                    return null
                }
                return LocalDateTimeUtil.parse(source, UniversalDateTime.DATE_TIME_FORMAT)
            }
        }
    }
}
